File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports Models
Imports DevExpress.Web.Mvc
Imports DevExpress.Web

Namespace Controllers
	<HandleError> _
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult

			ViewData("Html") = String.Empty
			Return View(New EditHtmlText())
		End Function

		<ValidateInput(False)> _
		Public Function FeaturesPartial() As ActionResult
			Return PartialView("HtmlEditor")
		End Function

		 <HttpPost> _
		 Public Function DownloadFile() As FileContentResult
			Dim html As String = HtmlEditorExtension.GetHtml("htmlEdit")

			Dim result As New FileContentResult(System.Text.Encoding.UTF8.GetBytes(html), "text/html")
			result.FileDownloadName = "myFile.html"
			Return result
		 End Function

		Public Function FileUploaded() As ActionResult
			UploadControlExtension.GetUploadedFiles("uc", New UploadControlValidationSettings() With {.AllowedFileExtensions = New String() { ".html", ".htm", ".txt" }}, AddressOf FileUploadComplete)
			Return Nothing
		End Function

		Protected Sub FileUploadComplete(ByVal sender As Object, ByVal e As FileUploadCompleteEventArgs)
			If e.UploadedFile.IsValid Then
				e.CallbackData = System.Text.Encoding.UTF8.GetString(e.UploadedFile.FileBytes)
			End If
		End Sub
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Models;
using DevExpress.Web.Mvc;
using DevExpress.Web;

namespace Controllers {
    [HandleError]
    public class HomeController : Controller {
        public ActionResult Index() {

            ViewData["Html"] = String.Empty;
            return View(new EditHtmlText());
        }

        [ValidateInput(false)]
        public ActionResult FeaturesPartial() {
            return PartialView("HtmlEditor");
        }

         [HttpPost]
        public FileContentResult DownloadFile() {
            String html = HtmlEditorExtension.GetHtml("htmlEdit");

            FileContentResult result = new FileContentResult(System.Text.Encoding.UTF8.GetBytes(html), "text/html");
            result.FileDownloadName = "myFile.html";
            return result;
        }

        public ActionResult FileUploaded() {
            UploadControlExtension.GetUploadedFiles("uc", new UploadControlValidationSettings() {
                AllowedFileExtensions = new String[] { ".html", ".htm", ".txt" }
            }, FileUploadComplete);
            return null;
        }

        protected void FileUploadComplete(object sender, FileUploadCompleteEventArgs e) {
            if (e.UploadedFile.IsValid) {
                e.CallbackData = System.Text.Encoding.UTF8.GetString(e.UploadedFile.FileBytes);
            }
        }
    }
}
